osquery/core/init.cpp

```diff
@@ -689,4 +690,29 @@ void Initializer::shutdownNow(int retcode) {
   platformTeardown();
   _Exit(retcode);
 }
+
+void Initializer::attachEvents() {
```

Comment:
I do not think this is a good idea because we want to keep components isolated. Right now `Initializer` has a lot of dependencies and this risks making events dependent on `Initializer` (the `osquery_core_init` CMake target). Is there a strong reason for bringing this into `Initializer`?